src/DatabaseLibrary/query.py

```diff
@@ -66,20 +68,24 @@ def query(self, selectStatement, sansTran=False, returnAsDict=False):
                 mappedRows = []
                 col_names = [c[0] for c in cur.description]
 
+
                 for rowIdx in range(len(allRows)):
                     d = {}
                     for colIdx in range(len(allRows[rowIdx])):
                         d[col_names[colIdx]] = allRows[rowIdx][colIdx]
                     mappedRows.append(d)
-                return mappedRows
+                return 'DONE',mappedRows
 
-            return allRows
+            return 'DONE',allRows
+        except  Exception as Err:
+            logger.info('Error: %s' % Err)
```

Comment:
In this way you won't stop execution but if an exception occurs during db connection I would not proceed in what am I doing.

Moreover it looks like an error probably should be
```suggestion
            logger.error('Error: %s' % Err)
```

Reply:
in case of connection problems, the exception is handled in the other class.
Modified the queries in this way, it is possible to manage even the negative tests, not only the positive ones, using robots through the status that is output.